File: gui_apps/PyDoodle/step05/step05.py
"""
// ===================================================================================
// step05.py: adds ability to change color & thickness of the squiggle
//  application handles left mouse press & drag, right mouse press events
//   - draws a squiggle in the client window when left mouse is pressed & dragged
//   - if ctrl + left mouse press - display dialog to set new pen thickness
//   - if ctrl + right mouse press - display dialog to set new pen thickness
//     (NOTE: Ctrl + mouse clicks is perhaps not the best way to design a GUI.
//      We are using this method as we have not yet introduces menus & toolbars.
//      This flaw will be corrected in steps where these GUI elements are introduced)
//
// Tutorial - PyQt5 Doodle Application
// Based on a similar tutorial for Borland ObjectWindows Library (OWL) 
// @author: Manish Bhobe
// My experiments with the Qt Framework. Use at your own risk!!
// =====================================================================================
"""
import sys
import os
import pathlib
from PyQt5.QtCore import *
from PyQt5.QtGui import *
from PyQt5.QtWidgets import *
from mainWindow import *
import darkdetect

sys.path.append(os.path.join(pathlib.Path(__file__).parents[1], 'common'))
import mypyqt5_utils as utils


def main():
    app = QApplication(sys.argv)
    app.setStyle("Fusion")

    if darkdetect.isDark():
        utils.setDarkPalette(app)

    mainWindow = MainWindow()
    mainWindow.show()

    sys.exit(app.exec_())


if __name__ == "__main__":
    main()

File: gui_apps/Doodle/step05/step05.cc
// ============================================================================
// step05.cc: drawing single lines in window, but with their separate attributes
// of line width & color.
// 	- double left mouse click - set line width
//  - double right mouse click - set line color
//    (NOTE: this is an absolutely horrible GUI design!! Purpose of this code is not
//     to present a good GUI design, rather to be able to set line width & color.
//     This flaw will be corrected in further tutorials).
//
// Tutorial - Qt Scribble Application
// Based on a similar tutorial for Borland ObjectWindows Library (OWL)
// Created by Manish Bhobe.
// ===========================================================================

#include "DrawWindow.h"
#include "common.hxx"
#include <QApplication>
#include <QtGui>

int main(int argc, char **argv)
{
   QApplication app(argc, argv);
   app.setStyle("Fusion");

#ifdef Q_OS_WINDOWS
   if (windowsDarkThemeAvailable() && windowsIsInDarkTheme())
     setWinDarkPalette(&app);
#endif

   // create the GUI
   DrawWindow mainWindow;
   mainWindow.show();

   return app.exec();
}
